app/serializers/song_index_serializer.rb

```diff
@@ -1,5 +1,6 @@
 class SongIndexSerializer < ActiveModel::Serializer
-  attributes :singer, :name, :genre, :rating
+  attributes :singer, :name, :genre, :rating, :views
+  belongs_to :user, serializer: UserSerializer
```

Comment:
Did you test it? I see the cyclical problem. `SongIndex` loads `User`, and `User` loads `SongIndex`. You should create new serializers for solving this problem)